src/ui/public/visualize/visualization.js

```diff
@@ -126,6 +126,7 @@ uiModules
       });
 
       $scope.$on('$destroy', () => {
+        resizeChecker.off('resize');
```

Comment:
Looks like ResizeChecker has a destroy method that internally removes all event handlers. I switched the on destroy event handler to call the ResizeChecker destroy method. Since the ResizeChecker is created outside of that if block, then it can be destroyed outside of that if block.